recipe/meta.yaml

```diff
@@ -8,43 +8,80 @@ package:
 
 source:
   url: https://github.com/ec-jrc/pyPoseidon/archive/{{ version }}.tar.gz
-  sha256: db16f8215bbd534b2c0bce5617894aebdecd76b88016618ba8ce220dca75005e
+  sha256: d2eba8d0710e17b57c7578a4a0433b8e1451b8f19847a36343a0fe57e8b085c7
 
 build:
   noarch: python
   number: {{ build }}
-  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
 
-requirements:
-  build:
-  host:
-    - pip
-    - python >=3.6
-    - setuptools
-    - poetry
-    - poetry-core
-  run:
-    - python >=3.6
-    - mayavi
-    - pyresample
-    - pygeos >=0.7.1
-    - geoviews
-    - rasterio
-    - jigsaw =0.9.12
-    - hvplot
-    - descartes
-    - cfgrib <0.9.9
-    - moviepy
-    - colorlog
-    - f90nml
-    - zarr
+outputs:
+  - name: pyposeidon-base
+    script: build_base.sh
+    build:
+      noarch: python
+    requirements:
+      build:
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+      host:
+        - pip
+        - python =3.8
+      run:
+        - python =3.8
+        - pyresample
+        - pygeos >=0.10
+        - rasterio
+        - rioxarray
+        - jigsaw
+        - cfgrib
+        - colorlog
+        - f90nml
+        - zarr
+        - gmsh >=4.8.4
+        - python-gmsh
+        - tqdm
+        - xarray
+        - pandas
+        - geopandas >=0.10.1
+        - numpy
+        - psutil
+        - requests
+        - scipy
+        - netCDF4
+        - dask
+        - shapely =1.7.1
+        - pydap >=3.3.0
+
+    test:
+      imports:
+
+  - name: pyposeidon
+    build:
+      noarch: python
+    requirements:
+      host:
+        - pip
+        - python =3.8
+      run:
+        - python =3.8
+        - {{ pin_subpackage("pyposeidon-base", exact=True) }}
+        - holoviews
+        - geoviews
+        - hvplot
+        - ipympl
+        - itkwidgets
+        - mayavi
+        - matplotlib-base >=3.3
+        - cartopy <=0.20
+        - ffmpeg
+        - spatialpandas
+
+    test:
+      imports:
```

Comment:
And here as well.

Reply:
I had to remove the test import because the build fails with it. I don't know why. 

There are a number of warnings of `conda.models.version:get_matcher` when solving the env and then it finds conflicts.

Reply:
Then unfortunately i feel like there is still something wrong with the recipe.

Reply:
I'm going to try a few things in: https://github.com/conda-forge/pyposeidon-feedstock/pull/3

Reply:
some of the hard dependencies require libgl. This may be a problem when you want to use this in a headless configuration. You should reconsider them

Reply:
I think running the tests is important, otherwise you've never actually solved the runtime environment.

Ultimately, I think that the exact version of shapely you want for conda-forge is just too old.
https://github.com/conda-forge/pyposeidon-feedstock/pull/3/files

Seems to converge if you loosen the pin.

Goodluck with your package!

Reply:
Hi @hmaarrfk.

Apologies for the delayed response but last week I was on leave. 

First I need to extend our gratitude for all your hard work during Sunday no less trying to resolve this issue. We can confirm that your modified `meta.yaml` works. 

However, it is very strange that although `conda build recipe` finishes correctly even with shapely =1.7.1, executing `build-locally.py` doesn't. This indicates an issue with the CI rather than the recipe itself.

The 1.7.1 version of `shapely` is not old. It is just the previous version. 

We'll tree to troubleshoot the issue but if you have any ideas on what to test regarding the setup of the container please advice.

Reply:
my guess is tgat locally you are pulling in packages from an other channel

Reply:
I am using 

```
channels:
  - conda-forge
  - defaults

anaconda_upload: false
channel_priority: strict
```

Reply:
To be precise, none of the versions work on my mac (?) but your recipe works on Linux including the version that pins `shapely` to 1.7.1.

Reply:
We recently removed `defaults` from the isolated build process

Reply:
Removing `defaults` from `.condarc` doesn't change the behaviour.

Reply:
I'm not too sure then. If you use `tee` i can try to help debug. But without a log it is hard.